.gitlab-ci.yml

```diff
@@ -841,45 +840,51 @@ test_backend_integration:
       };
       trap handle_exit EXIT
 
-    - INTEGRATION_TEST_SUITE=$(integration/extra/release_tool.py --select-test-suite || echo "all")
-    - echo Running backend-tests suite $INTEGRATION_TEST_SUITE
-    - cd integration/backend-tests/
+    - INTEGRATION_TEST_SUITE=$(integration/extra/release_tool.py --select-test-suite)
```

Comment:
@kacf This has been brought up couple of times, and it was also taken into consideration when designing this task.

The implementation detail here is that `--select-test-suite` option in release tool relies on all repositories been checked out in order to decide if this PR needs open, enterprise, or all tests. IMO, this option does not belong to the release tool and the logic should be moved completely to `integration-test-bot`. And by doing so, we could achieve what you are asking for, which will be much cleaner (i.e. jobs won't get started at all if they will end up in NOOP).

Do you want me to follow-up on a JIRA task for this? From my side, I can live with it :)

Reply:
> IMO, this option does not belong to the release tool and the logic should be moved completely to `integration-test-bot`.

Agree.

> Do you want me to follow-up on a JIRA task for this? From my side, I can live with it :)

I think I would like a task for this actually, although it's not urgent. Right now I don't feel confident that launching a job will really test everything, because it relies on this "smart script" for autodetection. It would be better if this could be overridden by parameters.

There is also an efficiency benefit, that we don't have to launch a slave at all for the job which is ultimately skipped.